File: 202312/1758.go
func minOperations(s string) int {
	return min(f(s, true), f(s, false))
}

func f(s string, curr bool) int {
	n := len(s)
	ans := 0
	for i := 0; i < n; i++ {
		if curr && s[i] == '1' || !curr && s[i] == '0' {
			ans++
		}
		curr = !curr
	}
	return ans
}

File: 202603/1758.py
class Solution:
    def minOperations(self, s: str) -> int:
        def f(c):
            ans = 0
            for i in range(len(s)):
                if c != s[i]:
                    ans += 1
                c = "1" if c == "0" else "0"
            return ans

        return min(f("1"), f("0"))
